client/src/pages/Client/index.tsx

```diff
@@ -1,95 +1,220 @@
+        setLoading(false);
+      }
+    };
+
+    getData();
+  }, []);
+
+  // fun delete item
+  const handleDelete = async (idItem: number) => {
+    try {
+      const dataItemDel = await axios.delete(`/api/v1/jobs/${idItem}`);
+      setMessage(dataItemDel.data.msg);
+      const filterJobsDelete = jobsUnoccupied.filter((job) => job.id !== idItem);
+      setJobsUnoccupied(filterJobsDelete);
+    } catch (err) {
+      setLoading(false);
+    }
+    setOpenAlert(true);
+  };
+
+  // fun Accipt proposal
+  const acceptPropsal = async (proposalID: number, jobID: number) => {
+    try {
+      const getProposalItem = await axios.patch(`/api/v1/proposals/${proposalID}`);
+
+      setAccept(getProposalItem.data.msg);
+      const jobAccepted = jobsUnoccupied.find((job) => job.id === jobID);
+      if (jobAccepted) {
+        const filterProposal = jobsUnoccupied.filter((job) => job.id !== jobID);
+        setJobsUnoccupied(filterProposal);
+        setJobsOccupied([...jobsOccupied, jobAccepted]);
+      }
+    } catch (err) {
+      setLoading(false);
+    }
+    setOpenAlertproposal(true);
+  };
+
+  // pending jobs for most jobs
+  let pendingJobs: React.ReactElement | null = null;
+
+  // accepted jobs for most jobs
+  let acceptedJobs: React.ReactElement | null = null;
+  // component while rendering data
+
+  if (loading) {
+    pendingJobs = (
+      <div className="spinner">
+        <CircularProgress color="inherit" />
+        {' '}
+      </div>
+    );
+  }
+
+  if (!jobsUnoccupied.length) { // pending jobs for most jobs
+    pendingJobs = <h2>No jobs found</h2>;
+  } else {
+    pendingJobs = (
+      // component while rendering data on pending jobs
+      <>
+        {jobsUnoccupied.map((job) => (
+          <JobCard job={job} deleteItem={handleDelete} key={job.title}>
+            <Accordion disabled={false}>
+              <AccordionSummary>
+                <div className="budget-proposal-section budget-proposal-client ">
+                  <div className="proposals">
+                    proposals:
+                    <span>
+                      {job.proposals.length}
+                    </span>
+                  </div>
+                  <div className="budget">
+                    budget:
+                    <span>
+                      $
+                      {job.budget}
+                    </span>
+                  </div>
+                </div>
+              </AccordionSummary>
+              <AccordionDetails>
+                {job.proposals.map((proposal) => <ProposalJob proposal={proposal} acceptProposal={acceptPropsal} />)}
+
+              </AccordionDetails>
+            </Accordion>
+
+          </JobCard>
+        ))}
+      </>
+    );
+  }
+
+  if (!jobsOccupied.length) { // accepted jobs for most jobs
+    acceptedJobs = <h2>No jobs found</h2>;
+  } else {
+    acceptedJobs = (
+      // component while rendering data on accepted jobs
+      <>
+        {jobsOccupied.map((job) => (
```

Comment:
I can see that we repeat ouselves in both in `occupeied` and `unoccupied` jobs,
May be we can create a component to handle this